independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/ParameterConverter.java

```diff
@@ -11,4 +11,9 @@ public interface ParameterConverter {
     default void init(ParamConverterProviders deployment, Class<?> rawType, Type genericType, Annotation[] annotations) {
 
     }
+
+    // TODO: this API method may be too limiting
+    default boolean isForSingleObjectContainer() {
```

Comment:
The name of the method is not very descriptive to me. 
Other than that, I would prefer adding a new implementation `ParameterCollectionConverter` instead of this method. But this is just a matter of taste.

Reply:
You would prefer `ParameterCollectionConverter` and checking the instance? That doesn't sound better to me :)

Reply:
Still think the method name is not very descriptive. Anyway, let's move forward.